### PR TITLE
feat: Upgrade the MongoDB.Driver to v2.5

### PR DESCRIPTION
**Pull request by rukai-kooboo**
- Upgrade the MongoDB SDK to version 2.5
- Upgrade assembly version to 1.1
- Update the gulp version and nuget-runner package
- Update nuget.exe to latest version to support appveyor
- Include the DNSclient from Nuget

### CODE REVIEW

**Comment by rukai-kooboo** [comment]
Hi @kspearrin 
It is a great job that you have built this project. We use it in our project with MongoDB v2.5 SDK. Will you spend a few minutes to help review the upgrade?
Thanks!

File: src/MongoRiver.Tests/packages.config [hunk 1]
```diff
@@ -1,9 +1,12 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="MongoDB.Bson" version="2.0.0" targetFramework="net451" />
-  <package id="MongoDB.Driver" version="2.0.0" targetFramework="net451" />
-  <package id="MongoDB.Driver.Core" version="2.0.0" targetFramework="net451" />
+  <package id="DnsClient" version="1.0.7" targetFramework="net451" />
```
Comment (kspearrin) [inline src/MongoRiver.Tests/packages.config:3]: Why were all these other dependencies added? I don't see them listed here? https://www.nuget.org/packages/MongoDB.Bson/

Comment (rukai-kooboo) [reply src/MongoRiver.Tests/packages.config:3]: Do you mean DnsClient? Checked, it is depended by:
MongoDB.Driver.Core.2.5.0

For other packages, they are included together when installing MongoDB.Driver 2.5.0

**Review by kspearrin — APPROVED** [review]

**Comment by kspearrin** [comment]
Please tr version 1.1.0 on nuget now and let me know if it's good. Thanks.

**Pull request merged**